test-guru/db/migrate/20191023080822_add_tests_category_id_foreign_key.rb

```diff
@@ -0,0 +1,5 @@
+class AddTestsCategoryIdForeignKey < ActiveRecord::Migration[6.0]
+  def change
+    add_reference :tests, :categories, index: true, foreign_key: true
```

Comment:
```suggestion
    add_reference :tests, :category, foreign_key: true
```
- посмотри примеры в документации: имя ссылки нужно писать в единственном числе
- воторой момент - индекс по умолчанию создается при использовании метода, явно эту опцию имеет смысл писать только с параметром "фолс", если не нужно создавать индекс
- ты не убрал предыдущее поле из таблицы